File: src/Presence/HexMaster.Attendr.Presence.Abstractions/Dtos/PresentationToRateDto.cs
namespace HexMaster.Attendr.Presence.Abstractions.Dtos;

/// <summary>
/// DTO containing presentation information for rating.
/// </summary>
/// <param name="PresentationId">The unique identifier of the presentation.</param>
/// <param name="Title">The title of the presentation.</param>
/// <param name="Abstract">The abstract/description of the presentation.</param>
/// <param name="Room">The name of the room where the presentation is held.</param>
/// <param name="StartDateTime">The start date and time of the presentation.</param>
/// <param name="EndDateTime">The end date and time of the presentation.</param>
/// <param name="Speakers">The collection of speakers for the presentation.</param>
/// <param name="Topics">The collection of topics associated with the presentation.</param>
public record PresentationToRateDto(
    Guid PresentationId,
    string Title,
    string Abstract,
    string Room,
    DateTime StartDateTime,
    DateTime EndDateTime,
    IReadOnlyCollection<PresentationSpeakerDto> Speakers,
    IReadOnlyCollection<PresentationTopicDto> Topics
);

File: src/App/src/app/shared/models/presentation-to-rate-dto.ts
import { SpeakerDto } from './speaker-dto';

export interface PresentationTopicDto {
    key: string;
    name: string;
}

export interface PresentationToRateDto {
    presentationId: string;
    title: string;
    abstract: string;
    room: string;
    startDateTime: string;
    endDateTime: string;
    speakers: SpeakerDto[];
    topics: PresentationTopicDto[];
}
